File: Data Structures & Algorithms/max-water-container/submission-40.cs
public class Solution {
    public int MaxArea(int[] heights) {
        var res = 0;
        var l = 0;
        var r = heights.Length - 1;

        while (l < r) {
            var area = Math.Min(heights[l], heights[r]) * (r - l);
            res = Math.Max(area, res);
            if (heights[l] < heights[r]) {
                l++;
            } else {
                r--;
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-40.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        maxF = 0
        count = {}
        l = 0
        res = 0
        for r in range(len(s)):
            count[s[r]] = count.get(s[r], 0) + 1
            maxF = max(count[s[r]], maxF)

            while r - l + 1 - maxF > k:
                count[s[l]] -= 1
                l += 1
            res = max(res, r - l + 1)
        return res